public/content/documentation/web/page-level/form.md

```diff
@@ -1,80 +1,603 @@
+   - WHEN I use the tab key to move focus to a form field
+      - I SEE focus is visually indicated
+      - I SEE the focus moves in a logical order
 
 2. Desktop screenreader
-
-   - WHEN I use a desktop screenreader (NVDA, JAWS, VoiceOver) AND
-
-   - I use the tab key to enter the web browser window
-
-   - I HEAR It is discoverable with screenreader shortcuts as header/banner landmark
-
-   - I HEAR It typically contains the name and primary navigation of the website
+   - WHEN I use a desktop screenreader (NVDA, JAWS, VoiceOver) AND 
+   - I use the tab key to move focus to inputs inside the form
+     - I HEAR If multiple forms are present (Ex: Search, Sign in, Newsletter subscription), the form must have a name
+     - I HEAR It is discoverable with screenreader shortcuts as a form landmark along with its name
+     - I HEAR the screenreader switch from browse shortcuts to forms shortcuts
 
 3. Mobile screenreader
-
    - WHEN I use a mobile screenreader (Talkback, VoiceOver) AND
-
-   - I swipe to focusable elements in the header
-
-   - I HEAR It is discoverable with screenreader shortcuts as header/banner landmark
-
-   - I HEAR It typically contains the name and primary navigation of the website
-
+   - I swipe to focus on inputs inside the form
+     - I HEAR If multiple forms are present (Ex: Search, Sign in, Newsletter subscription), the form must have a name
+     - I HEAR It is discoverable with screenreader shortcuts as a form landmark along with its name
+     - I HEAR the screenreader switch from browse shortcuts to forms shortcuts
 
 Full information: [https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form](https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form)
 
 ## Developer Notes
 
-### Name
-
-- Typically doesn’t have a name or description since there must be only one instance per page.
+### Do not auto focus inputs
+
+Automatically moving focus to an input field is very confusing for people using assistive technology.
+
+### Group inputs
+
+Use `fieldset` and `legend` to group related fields, such as:
+- Sign in
+- Shipping address
+- Payment information
+
+### Error handling
+
+- Individual inputs must have programmatically described errors read by the screen reader on focus.
+- For long forms, list all errors in an alert with links back to each invalid input on submission attempts.
+
+### Field width affordance
+
+- Inputs should visually reflect the expected data length.
+    - Middle initial: 1 character
+    - State abbreviation: 2 characters
+    - Zip code: 5 characters
+    - PIN: depends on expected digits
+
+### Layout: Stack inputs vertically
+
+Avoid multiple columns:
+- Vertical scrolling is more accessible and expected.
+- Screen magnifiers make it difficult to find right-column fields.
+- Submit buttons should not be placed in sidebars only.
+
+### Use autocomplete
+
```

Comment:
Could we include a line under this heading supporting this, like "Utilizing autocomplete allows allows users automated assistance in filling out form field values." or something like it and link to MDN's Autocomplete documentation on the word "autocomplete": https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/autocomplete 

as an aside, autocomplete and inputmode are both on my list of wanting to integrate more into our magentaA11Y documentation!